engine/gui/controller.py

```diff
@@ -35,6 +35,15 @@ def on_key_press(self, symbol: int, modifiers: int) -> None:
     def on_key_release(self, symbol: int, modifiers: int) -> None:
         if symbol == arcade.key.ENTER:
             self._activate_current_button()
+        elif symbol == arcade.key.RIGHT and self.selected_button.right is not None:
```

Comment:
You'll want to tweak the logic here, because if `self.selected_button.right` is `None` then the if block will continue falling through to later checks. Instead it's cleaner to have the handler for each button type as an isolated bit of code.

Something like this:
```python
elif symbol == arcade.key.RIGHT:
  if self.selected_button.right is not None:
    ...
```
The value of this approach is more about making code easier to reason about as it changes and gets more complex. With a simple condition like `symbol == arcade.key.RIGHT` the reader knows "OK, this is where _all_ the logic related to right key presses is handled.
It's also less error-prone because if someone rights another clause further down this giant `if` block that just happens to be true, we could accidentally trigger it when someone presses the right key.